internal/transcribe/google_transcriber.go

```diff
@@ -0,0 +1,69 @@
+package transcribe
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	speech "cloud.google.com/go/speech/apiv1"
+	"cloud.google.com/go/storage"
+)
+
+type googleTranscriber struct{}
+
+func (gt *googleTranscriber) process(ctx context.Context, filename string, opts TranscribeOptions) (string, error) {
+	client, err := createSpeechClient(ctx, opts)
+	if err != nil {
+		return "", err
+	}
+	defer func(client *speech.Client) {
+		err := client.Close()
+		if err != nil {
+			fmt.Printf("failed to close transcribe client: %v\n", err)
+		}
+	}(client)
```

Comment:
The fmt.Printf is a bit strange to me

The defer is called before returning the value

So you could do something like this


```suggestion
	defer func(client *speech.Client) {
		errCloser := client.Close()
		if errCloser != nil {
			err = errors.Join(err, fmt.Errorff("failed to close transcribe client: %w", errCloser)
		}
	}(client)
```

Reply:
The code might need more rewriting of course.

Another solution is to extract the code calling everything needed before the defer in a lamda function, then call the code currently in the defer after the lamda function

Reply:
I'm a little confused, doesn't the actual execution of the deferred function happen just before the containing function returns?

Your code is great by the way, I didn't know about errors.Join

Reply:
This article explains a bit

https://trstringer.com/golang-deferred-function-error-handling/

Here is an article about using defer plus errors.Join

https://wstrm.dev/posts/errors-join-heart-defer/